samples/batch.js

```diff
@@ -40,12 +38,21 @@ async function createQueryPartitions(
   // Gets a reference to a Cloud Spanner instance and database
   const instance = spanner.instance(instanceId);
   const database = instance.database(databaseId);
-  const transaction = database.batchTransaction(identifier);
+  const [transaction] = await database.createBatchTransaction();
 
   const query = 'SELECT * FROM Singers';
 
   const [partitions] = await transaction.createQueryPartitions(query);
   console.log(`Successfully created ${partitions.length} query partitions.`);
+
+  let row_count = 0;
+  for (const partition of partitions) {
```

Comment:
Would it make sense to let this sample read the different partitions in parallel instead of sequentially, as that is the intention behind batch transactions? The read actions could then be added to an array of promises, and the sample would then wait for all the read operations on the different partitions to finish.

Reply:
Thanks @olavloite, I think that's a good idea. I modified it to use promises. PTAL.